activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -230,7 +230,15 @@ def column_for_attribute(name)
     #   person.respond_to(:nothing) # => false
     def respond_to?(name, include_private = false)
       return false unless super
-      name = name.to_s
+
+      case name
```

Comment:
You're doing an awesome work, but I'm not the one that has to decide if the benefit is not worth the cost on terms of purity or readability, anyway why don't you call respond_to? in those places passing a "string".freeze as argument instead a symbol